cmake/modules/RootMacros.cmake

```diff
@@ -282,6 +282,95 @@ function(ROOT_GENERATE_DICTIONARY dictionary)
     set(CMAKE_INSTALL_LIBDIR ${CMAKE_CURRENT_BINARY_DIR})
     set(libprefix "")
   endif()
+  
+  
+  if((CMAKE_PROJECT_NAME STREQUAL ROOT) AND (TARGET ${ARG_MODULE}))
+    set(incdirs)
+
+    # for a time been, roocling makes strange error
+    list(APPEND incdirs ${CMAKE_BINARY_DIR}/include)
+    
+    if(TARGET ${ARG_MODULE})
+       get_target_property(target_incdirs ${ARG_MODULE} INCLUDE_DIRECTORIES)
+       foreach(dir ${target_incdirs})
+           list(APPEND incdirs ${dir})
+       endforeach()
+    endif()
+
+    foreach(dir ${ARG_INCLUDES})
+        list(APPEND incdirs ${dir})
+    endforeach()
+
+    list(REMOVE_DUPLICATES incdirs)
+    
+    foreach(d ${incdirs})
+       list(APPEND includedirs -I${d})
+    endforeach()
+    
+    set(pureincdirs ${incdirs})
+    
+    set(headerfiles)
+    set(_list_of_header_dependencies)
+    foreach(fp ${ARG_UNPARSED_ARGUMENTS})
+      if(${fp} MATCHES "[*?]") # Is this header a globbing expression?
+        file(GLOB files inc/${fp} ${fp}) # Elements of ${fp} have the complete path.
+        foreach(f ${files})
+          if(NOT f MATCHES LinkDef) # skip LinkDefs from globbing result
+            set(add_inc_as_include On)
+            string(REGEX REPLACE "^${CMAKE_CURRENT_SOURCE_DIR}/inc/" "" f_no_inc ${f})
+            list(APPEND headerfiles ${f_no_inc})
+            list(APPEND _list_of_header_dependencies ${f})
+          endif()
+        endforeach()
+      else()
+        if(IS_ABSOLUTE ${fp})
+          set(headerFile ${fp})
+        else()
+          set(incdirs_in_build)
+          set(incdirs_in_prefix ${headerdirs_dflt})
+          foreach(incdir ${incdirs})
+            if(NOT IS_ABSOLUTE ${incdir})
+              list(APPEND incdirs_in_build ${incdir})
+            else()
+              list(APPEND incdirs_in_prefix ${incdir})
+            endif()
+          endforeach()
+          if(incdirs_in_build)
+            find_file(headerFile ${fp}
+              HINTS ${incdirs_in_build}
+              NO_DEFAULT_PATH
+              NO_SYSTEM_ENVIRONMENT_PATH
+              NO_CMAKE_FIND_ROOT_PATH)
+          endif()
+          # Try this even if NOT incdirs_in_prefix: might not need a HINT.
+          if(NOT headerFile)
+            find_file(headerFile ${fp}
+              HINTS ${incdirs_in_prefix}
+              NO_DEFAULT_PATH
+              NO_SYSTEM_ENVIRONMENT_PATH)
+          endif()
+        endif()
+        if(NOT headerFile)
+          message(FATAL_ERROR "Cannot find header ${fp} to generate dictionary ${dictionary} for. Did you forget to set the INCLUDE_DIRECTORIES property for the current directory?")
```

Comment:
What happens if the file is generated at build time, but doesn't exist yet at config time?

Reply:
We do not used generated header for dictionaries

Reply:
I'd be in favor of adding a "FIXME: We do not consider files generated at build time as dictionaries do not use such files. Relax the condition if that changes".

Reply:
I guess, fatal error is ok. All headers which are provided for dictionary generation should exists at time when macro is executed

Reply:
The fatal error is for users, FIXMEs are for our-future-selves to avoid wasting time re-understanding what we did ;)

Reply:
AFAIK users use this function, too?

Reply:
`if(CMAKE_PROJECT_NAME STREQUAL ROOT)` in mind it is only for ROOT libraries. If it is not a case - that is right check? My intention was to introduce special handling **ONLY** for ROOT libraris

Reply:
Sorry I missed that. I'm okay with this change. I bet Vassil is okay now, too, given that this is only for ROOT building itself.

Reply:
As I said, I'd be in favor of adding a fixme with more technical developer info as proposed. If @linev thinks it is not useful then I am fine.